client/api/omni/specs/cluster_machine_config_patches.go

```diff
@@ -0,0 +1,166 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package specs
+
+import (
+	"errors"
+
+	"github.com/siderolabs/gen/xslices"
+	"gopkg.in/yaml.v3"
+)
+
+// MarshalJSON implements json.Marshaler interface.
+//
+// It represents compressed fields as uncompressed in the output.
+func (x *ClusterMachineConfigPatchesSpec) MarshalJSON() ([]byte, error) {
+	obj := x.CloneVT()
+
+	patches, err := obj.GetUncompressedPatches()
+	if err != nil {
+		return nil, err
+	}
+
+	obj.Patches = patches
+	obj.CompressedPatches = nil
+
+	return jsonMarshaler.Marshal(obj)
+}
+
+// UnmarshalJSON implements json.Unmarshaler interface.
+func (x *ClusterMachineConfigPatchesSpec) UnmarshalJSON(data []byte) error {
+	return unmarshalJSONMultiple(x, data, GetCompressionConfig())
+}
+
+// MarshalYAML implements yaml.Marshaler interface.
+func (x *ClusterMachineConfigPatchesSpec) MarshalYAML() (any, error) {
+	patches, err := x.GetUncompressedPatches()
+	if err != nil {
+		return nil, err
+	}
+
+	contents := xslices.Map(patches, func(patch string) *yaml.Node {
+		style := yaml.FlowStyle
+		if len(patch) > 0 && (patch[0] == '\n' || patch[0] == ' ') {
+			style = yaml.SingleQuotedStyle
+		}
+
+		return &yaml.Node{
+			Kind:  yaml.ScalarNode,
+			Tag:   "!!str",
+			Style: style,
+			Value: patch,
+		}
+	})
+
+	return &yaml.Node{
+		Kind: yaml.MappingNode,
+		Tag:  "!!map",
+		Content: []*yaml.Node{
+			{Kind: yaml.ScalarNode, Tag: "!!str", Value: "patches"},
+			{Kind: yaml.SequenceNode, Tag: "!!seq", Content: contents},
+		},
+	}, nil
+}
+
+// UnmarshalYAML implements yaml.Unmarshaler interface.
+func (x *ClusterMachineConfigPatchesSpec) UnmarshalYAML(node *yaml.Node) error {
+	type alias ClusterMachineConfigPatchesSpec // prevent recursion
+
+	aux := (*alias)(x)
+
+	return unmarshalYAMLMultiple(x, aux, node, GetCompressionConfig())
+}
+
+// GetUncompressedData returns the patches from the ClusterMachineConfigPatchesSpec, decompressing them if necessary.
+func (x *ClusterMachineConfigPatchesSpec) GetUncompressedData(opts ...CompressionOption) ([]Buffer, error) {
+	if x == nil {
+		return nil, nil
+	}
+
+	if x.CompressedPatches == nil {
+		buffers := make([]Buffer, 0, len(x.Patches))
+
+		for _, patch := range x.Patches {
+			buffers = append(buffers, newNoOpBuffer([]byte(patch)))
+		}
+
+		return buffers, nil
```

Comment:
`xslices.Map` can make it much shorter:

```suggestion
		return xslices.Map(x.Patches, func(patch string) []Buffer {
			return newNoOpBuffer([]byte(patch))
		})
```

Reply:
I kinda avoided to bring that package into the `specs` package (as it is public), but now that I checked, it was already used on `ClusterMachineConfigPatchesSpec.MarshalYAML` (I just moved this function, didn't change).

So I guess it makes sense to use it here as well now.